### ISSUE TITLE
charset false positive

### ISSUE THREAD

**Issue by HansHammel** [arrowrowe/textlint-rule-editorconfig#2]
charset gives false positives and reports utf8 to be ascii
how do i disable the rule? `"editorconfig/charset" : false,` has no effect.

**Comment by arrowrowe** [comment]
Hey @HansHammel, Thanks for reporting! I will take a look this weekend.

For now, does a `.textlintrc` as following work for you?

```javascript
{
  "rules": {
    "editorconfig": {
        "charset": false
    }
  }
}
```

**Comment by HansHammel** [comment]
no sir! tried that already. btw. other rules in the rc file are working fine.

**Comment by arrowrowe** [comment]
Sorry for the delay. Will try tonight.

**Comment by arrowrowe** [comment]
Seems that I did not handle `.textlintrc`, leaving charset and all other stuff configured by `.editorconfig`.

A quick workaround is to set charset to `false` in your `.editorconfig`.

**Comment by arrowrowe** [comment]
Hey @HansHammel would you like to checkout v1.0.3? [the `textlintrc` we mentioned before](https://github.com/arrowrowe/textlint-rule-editorconfig/issues/2#issuecomment-268705953) should work now.

diff I made: https://github.com/arrowrowe/textlint-rule-editorconfig/commit/cbd63e3b96032ce8e0b41abcf7932f820b84b897 rules/charset.js 
```diff
+  if (args.option.charset === false) {
+    return;
+  }
```

**Comment by arrowrowe** [comment]
Also, since we use [JsChardet](https://github.com/aadsm/jschardet) to detect the charset, maybe you could report your false alarm there if confirmed?

**Comment by HansHammel** [comment]
```
"editorconfig": {
    "charset": false
}
```
is working, just like the "no entry" case, but every other value like true, "utf8" or "UTF-8" leads to 
```
TypeError: Cannot read property 'toLowerCase' of null
at C:\a-project\AUTHORS.md
    at module.exports (C:\a-project\node_modules\textlint-rule-editorconfig\rules\charset.js:14:32)
    at rules.forEach (C:\a-project\node_modules\textlint-rule-editorconfig\index.js:15:29)
    at Array.forEach (native)
    at getConfig.then (C:\a-project\node_modules\textlint-rule-editorconfig\index.js:15:11)
    at tryCatcher (C:\a-project\node_modules\textlint-rule-editorconfig\node_modules\bluebird\js\release\util.js:16:23)
    at Promise._settlePromiseFromHandler (C:\a-project\node_modules\textlint-rule-editorconfig\node_modules\bluebird\js\release\promise.js:503:31)
    at Promise._settlePromise (C:\a-project\node_modules\textlint-rule-editorconfig\node_modules\bluebird\js\release\promise.js:560:18)
    at Promise._settlePromise0 (C:\a-project\node_modules\textlint-rule-editorconfig\node_modules\bluebird\js\release\promise.js:605:10)
    at Promise._settlePromises (C:\a-project\node_modules\textlint-rule-editorconfig\node_modules\bluebird\js\release\promise.js:684:18)
    at Async._drainQueue (C:\a-project\node_modules\textlint-rule-editorconfig\node_modules\bluebird\js\release\async.js:126:16)
    at Async._drainQueues (C:\a-project\node_modules\textlint-rule-editorconfig\node_modules\bluebird\js\release\async.js:136:10)
    at Immediate.Async.drainQueues (C:\a-project\node_modules\textlint-rule-editorconfig\node_modules\bluebird\js\release\async.js:16:14)
    at runCallback (timers.js:637:20)
    at tryOnImmediate (timers.js:610:5)
    at processImmediate [as _immediateCallback] (timers.js:582:5
```

**Comment by docwhat** [comment]
I'm not sure `jschardet` is the right package to be using because it just returns one answer.

What you really want is something that either returns a bunch of character sets so you can see if you have enough confidence that it is the editorconfig specified charset.

Alternatively, if there is a library where you can ask "is this file this charset" instead.

A bunch of my documents detect as `ascii` which is silly, since `ascii` is a subset of `utf-8'.

Another group of documents match as `windows-1252` just because they have the UTF-8 non-breaking space character in it.

**Comment by InertSloth** [comment]
I also had to abandon/disable this module for now. 

I was getting a ton of false positives for `windows-1252` likely because of the same reason as docwhat. Upwards of 15 files. And another file was saying `ascii` instead utf8.

I tried the fix with `"charset": false` from this thread since the docs did not say anything about disabling subrules. Doing so gave me this error when running:

```
$ pnpm run lint:text

> apathetic-idleon@0.0.1 lint:text /mnt/e/Code/apathetic-idleon.github.io
> textlint --cache src/content/docs/**/*.{md,mdx}

Unexpected error during file processing: TypeError: option._pattern is not a function
at /mnt/e/Code/apathetic-idleon.github.io/src/content/docs/mechanic/world-4/tome.md
    at /mnt/e/Code/apathetic-idleon.github.io/node_modules/.pnpm/textlint-rule-editorconfig@1.0.3/node_modules/textlint-rule-editorconfig/editorconfig/get-config.js:10:16
    at module.exports (/mnt/e/Code/apathetic-idleon.github.io/node_modules/.pnpm/textlint-rule-editorconfig@1.0.3/node_modules/textlint-rule-editorconfig/editorconfig/loop-root.js:10:5)
    at /mnt/e/Code/apathetic-idleon.github.io/node_modules/.pnpm/textlint-rule-editorconfig@1.0.3/node_modules/textlint-rule-editorconfig/editorconfig/get-config.js:9:3
    at tryCatcher (/mnt/e/Code/apathetic-idleon.github.io/node_modules/.pnpm/bluebird@3.3.4/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/mnt/e/Code/apathetic-idleon.github.io/node_modules/.pnpm/bluebird@3.3.4/node_modules/bluebird/js/release/promise.js:503:31)
    at Promise._settlePromise (/mnt/e/Code/apathetic-idleon.github.io/node_modules/.pnpm/bluebird@3.3.4/node_modules/bluebird/js/release/promise.js:560:18)
    at Promise._settlePromise0 (/mnt/e/Code/apathetic-idleon.github.io/node_modules/.pnpm/bluebird@3.3.4/node_modules/bluebird/js/release/promise.js:605:10)
    at Promise._settlePromises (/mnt/e/Code/apathetic-idleon.github.io/node_modules/.pnpm/bluebird@3.3.4/node_modules/bluebird/js/release/promise.js:684:18)
    at Async._drainQueue (/mnt/e/Code/apathetic-idleon.github.io/node_modules/.pnpm/bluebird@3.3.4/node_modules/bluebird/js/release/async.js:126:16)
    at Async._drainQueues (/mnt/e/Code/apathetic-idleon.github.io/node_modules/.pnpm/bluebird@3.3.4/node_modules/bluebird/js/release/async.js:136:10)
    at Async.drainQueues (/mnt/e/Code/apathetic-idleon.github.io/node_modules/.pnpm/bluebird@3.3.4/node_modules/bluebird/js/release/async.js:16:14)
    at process.processImmediate (node:internal/timers:485:21)
 ELIFECYCLE  Command failed with exit code 1.
```

You can find my project if you want to try it yourself, you'll obviously have to edit the `.textlintrc.json` in the root and run `pnpm run lint:text`.

https://github.com/apathetic-idleon/

\- 🦥
